src/convolution_api.cpp

```diff
@@ -198,39 +198,48 @@ extern "C" miopenStatus_t miopenGetConvolutionFindMode(const miopenConvolutionDe
 
 extern "C" miopenStatus_t
 miopenConvolutionCKBackwardWeightsGetWorkSpaceSize(const miopenAlphaBetaCase_t alpha_beta_case,
-                                                   miopenDataType_t data_type,
-                                                   size_t C,
-                                                   size_t K,
-                                                   size_t output_tensor_size,
+                                                   const miopenTensorDescriptor_t inputTensorDesc,
+                                                   const miopenTensorDescriptor_t outputTensorDesc,
                                                    size_t* buffer_size)
 {
+    MIOPEN_LOG_FUNCTION(alpha_beta_case, outputTensorDesc);
+    return miopen::try_([&] {
+        miopenDataType_t data_type = miopen::deref(outputTensorDesc).GetType();
+        size_t in_spatial_dims     = miopen::deref(inputTensorDesc).GetNumDims();
 
-    size_t byte_size = 0;
-    if(alpha_beta_case == BILINEAR || alpha_beta_case == SCALE ||
-       ((data_type == miopenHalf) && ((C & 1) != 0 || (K & 1) != 0 /* Test if odd*/)))
-    {
-        switch(data_type)
+        assert(in_spatial_dims == miopen::deref(outputTensorDesc).GetNumDims());
+
+        size_t C = std::get<1>(
+            miopen::GetNCDHW(in_spatial_dims, miopen::deref(inputTensorDesc).GetLengths()));
+        size_t K = std::get<1>(
+            miopen::GetNCDHW(in_spatial_dims, miopen::deref(outputTensorDesc).GetLengths()));
+        size_t output_tensor_size = miopen::deref(outputTensorDesc).GetElementSize();
+
+        size_t byte_size = 0;
+        if((alpha_beta_case == BILINEAR || alpha_beta_case == SCALE ||
+            ((data_type == miopenHalf) && ((C & 1) != 0 || (K & 1) != 0 /* Test if odd*/))))
         {
-        case miopenInt32:
-        case miopenFloat:
-        case miopenHalf:
-        case miopenBFloat16:
-        case miopenInt8:
-        case miopenFloat8:
-        case miopenBFloat8: byte_size = 4; break;
-        case miopenDouble: byte_size = 8;
+            switch(data_type)
```

Comment:
I believe there's a function somewhere that gives size for type.

Reply:
I couldn't find the function whose return byte_size based on size of accumulator.

Reply:
I was thinking of `GetTypeSize()` in `src/include/.../tensor.hpp`. It doesn't quite do what you're doing here because it returns the actual size in bytes and not 4 bytes for types < 4. 

You could take `GetTypeSize()` and round it to the next multiple of 4, but that IMO is too clever and less clear.